projects/plugins/jetpack/extensions/blocks/videopress/editor.js

```diff
@@ -435,3 +436,115 @@ addFilter(
 	'videopress/add-core-video-transform',
 	addVideoPressCoreVideoTransform
 );
+
+/**
+ * Organize the block attributes for the new videopress/video block
+ *
+ * @param {object} attributes        - core/video block attributes
+ * @param {object} defaultAttributes - default core/video block attributes
+ * @returns {object}                   The new attributes
+ */
+function getVideoPressVideoBlockAttributes( attributes, defaultAttributes ) {
+	const attrs = attributes || defaultAttributes;
+
+	// Update attributes names to match the new VideoPress Video block.
+	if ( attrs?.videoPressTracks ) {
+		attrs.tracks = attrs.videoPressTracks || [];
+		delete attrs.videoPressTracks;
+	}
+
+	if ( attrs?.isVideoPressExample ) {
+		attrs.isExample = attrs.isVideoPressExample || [];
+		delete attrs.isVideoPressExample;
+	}
+
+	return attrs;
+}
+
+/**
+ * Check whether the block is a VideoPress block istance,
+ * based on the passed attributes.
+ *
+ * @param {object} attributes - Block attributes.
+ * @returns {boolean} 	        Whether the block is a VideoPress block istance.
+ */
+const isVideoPressBlockBasedOnAttributes = attributes => {
+	const { guid, videoPressTracks, isVideoPressExample } = attributes;
+
+	// VideoPress block should have a guid attribute.
+	if ( ! guid?.length ) {
+		return false;
+	}
+
+	// VideoPress block should have a videoPressTracks array attribute.
+	if ( ! Array.isArray( videoPressTracks ) ) {
+		return false;
+	}
+
+	// VideoPress block should have a isVideoPressExample boolean attribute.
+	const attrNames = Object.keys( attributes );
+	if ( ! attrNames.includes( 'isVideoPressExample' ) || typeof isVideoPressExample !== 'boolean' ) {
+		return false;
+	}
+
+	return true;
+};
+
+const convertCoreVideoToVideoPressVideoBlock = createHigherOrderComponent( BlockListBlock => {
+	return props => {
+		const { block } = props;
+		const { name, attributes, clientId, __unstableBlockSource } = block;
+		const { replaceBlock } = useDispatch( blockEditorStore );
+
+		/*
+		 * We try to recognize core/video Jetpack VideoPress block,
+		 * based on some of its attributes.
+		 */
+		const isCoreVideoVideoPressBlock = isVideoPressBlockBasedOnAttributes( attributes );
+
+		const isVideoPressVideoBlockRegistered = getBlockType( 'videopress/video' );
+
+		const { available: isVideoPressVideoBlockAvailable } = getJetpackExtensionAvailability(
+			'videopress/video'
+		);
+
+		const isSimple = isSimpleSite();
+
+		const shouldConvertToVideoPressVideoBlock = !! (
+			name === 'core/video' && // Only auto-convert if the block is a core/video block
+			isVideoPressVideoBlockRegistered && // Only auto-convert if the VideoPress block is registered
+			isCoreVideoVideoPressBlock && // Only auto-convert if the block is a VideoPress block
+			isVideoPressVideoBlockAvailable && // Only auto-convert if the feature is available
```

Comment:
Is this the flag that will be active for beta users?

Reply:
Yes, indirectly, it is. A few conditions define the block (feature) availability. One of them is if the feature is defined as beat. Other ones can be the feature requires a paid plan, for instance.

Also, this value is defined dynamically. So, for instance, when the site is proxied, the beta features are enabled, meaning that this feature will be available. Quite tricky to follow.

Reply:
Got it, nice to know.